### PR TITLE
ADEN-3990 Refactor wikia.uap and wikia.not_uap events

### PR DESCRIPTION
**Pull request by fraszczakszymon**
Change the way we dispatch `wikia.uap` and `wikia.not_uap` events so we don't rely on real ad status (success/hop/collapse) but we detect whether UAP was loaded inside AdDetect module.

### CODE REVIEW

File: extensions/wikia/AdEngine/js/template/bfaa.js [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
 	'ext.wikia.adEngine.adHelper',
 	'ext.wikia.adEngine.provider.btfBlocker',
 	'ext.wikia.adEngine.slotTweaker',
-	'ext.wikia.adEngine.uapContext',
-	'ext.wikia.adEngine.utils.eventDispatcher',
 	'wikia.document',
 	'wikia.log',
 	'wikia.window',
 	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaa.js:12]: MAJOR Function has 9 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues
- CRITICAL 4 critical
- MAJOR 11 major
- MINOR 2 minor
- INFO 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [adDetect.js#L295](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L295): Change this condition so that it does not always evaluate to "false". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2583)
2. CRITICAL [adDetect.js#L295](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L295): Change this condition so that it does not always evaluate to "false". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2583)
3. CRITICAL [adDetect.js#L299](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L299): Change this condition so that it does not always evaluate to "false". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2583)
4. CRITICAL [adDetect.js#L304](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L304): Change this condition so that it does not always evaluate to "true". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2583)
5. MAJOR [evolve2.js#L47](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/evolve2.js#L47): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
6. MAJOR [evolve2.js#L53](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/evolve2.js#L53): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
7. MAJOR [adDetect.js#L95](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L95): Function has a complexity of 25 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
8. MAJOR [adDetect.js#L163](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L163): Function has a complexity of 29 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
9. MAJOR [adDetect.js#L229](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L229): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
10. MAJOR [adDetect.js#L303](https://github.com/Wikia/app/blob/ffcd7fc4d19e918ef96a337778bc2a562a7aba04/extensions/wikia/AdEngine/js/provider/gpt/adDetect.js#L303): Merge this if statement with the nested one. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)

**Comment by v3rth** [comment]
LGTM

**Comment by bulaszenko** [comment]
Checked works good

**Pull request merged**